src/ImpactX.cpp

```diff
@@ -33,6 +36,41 @@ namespace impactx
 
         // todo: if charge deposition and/or space charge are requested, require
         //       amr.n_cells from user inputs
+
+        // query input for warning logger variables and set up warning logger accordingly
+        warn_logger_control();
+
+    }
+
+    void ImpactX::warn_logger_control()
```

Comment:
To improve compile time, can you put this in a separate file please, e.g., `src/initialization/Warnings.cpp`? :)